DatadogSessionReplay/Sources/Recorder/ScreenChangeMonitor/TimeProvider.swift

```diff
@@ -0,0 +1,71 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+// MARK: - Overview
+//
+// Defines the time abstraction used by screen change monitoring and its live
+// implementation. `LiveTimeProvider` reads now from CACurrentMediaTime (monotonic)
+// and schedules using Dispatch timers to avoid clock-change skew.
+
+#if os(iOS)
+import QuartzCore
+
+internal protocol TimeProviderTask {
+    func cancel()
+}
+
+internal protocol TimeProvider {
+    associatedtype Task: TimeProviderTask
+
+    var now: TimeInterval { get }
+
+    func schedule(after interval: TimeInterval, _ action: @escaping () -> Void) -> Task
+}
+
+internal struct LiveTimeProvider: TimeProvider {
+    final class Task: TimeProviderTask {
+        private var timer: DispatchSourceTimer?
+
+        init(timer: DispatchSourceTimer) {
+            self.timer = timer
+        }
+
+        func cancel() {
+            timer?.cancel()
+            timer = nil
+        }
+
+        deinit {
+            cancel()
+        }
+    }
+
+    var now: TimeInterval {
+        CACurrentMediaTime()
+    }
+
+    private let queue: DispatchQueue
+
+    init(queue: DispatchQueue = .main) {
+        self.queue = queue
+    }
+
+    func schedule(after interval: TimeInterval, _ action: @escaping () -> Void) -> Task {
+        let timer = DispatchSource.makeTimerSource(queue: queue)
```

Comment:
sanity check/ We’re creating, resuming, handling, and canceling a `DispatchSourceTimer` every time something changes on screen (debounced at ~100ms). Do we know if this introduces any noticeable overhead?

Reply:
I took some measurements:
- Timer creation takes on average 0.0148 milliseconds (14.8 microseconds)

So I think we're good with this approach